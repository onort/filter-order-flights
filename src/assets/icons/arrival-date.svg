<svg width="19" height="20" viewBox="0 0 19 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M17.5703 3.07253H1.16663C0.5235 3.07253 0 3.60159 0 4.25156V18.1141C0 18.764 0.5235 19.2931 1.16663 19.2931H17.5703C18.2137 19.2931 18.737 18.764 18.737 18.1141V4.25156C18.737 3.60159 18.2137 3.07253 17.5703 3.07253ZM17.6764 18.1141C17.6764 18.173 17.6287 18.2213 17.5703 18.2213H1.16663C1.1083 18.2213 1.06058 18.173 1.06058 18.1141V7.61006H17.6764V18.1141ZM17.6764 6.53822H1.06058V4.25156C1.06058 4.19261 1.1083 4.14437 1.16663 4.14437H17.5703C17.6287 4.14437 17.6764 4.19261 17.6764 4.25156V6.53822ZM14.9895 2.35805C15.2825 2.35805 15.5198 2.11817 15.5198 1.82213V0.53592C15.5198 0.239878 15.2825 0 14.9895 0C14.6966 0 14.4592 0.239878 14.4592 0.53592V1.82213C14.4592 2.11817 14.6966 2.35805 14.9895 2.35805ZM3.74744 2.35805C4.04037 2.35805 4.27772 2.11817 4.27772 1.82213V0.53592C4.27772 0.239878 4.04037 0 3.74744 0C3.45451 0 3.21715 0.239878 3.21715 0.53592V1.82213C3.21715 2.11817 3.45451 2.35805 3.74744 2.35805Z" fill="#BDC0C8"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M9.5 9L10.1125 9.6125L7.6625 12.0625H13V12.9375H7.6625L10.1125 15.3875L9.5 16L6 12.5L9.5 9Z" fill="#BDC0C8"/>
</svg>
