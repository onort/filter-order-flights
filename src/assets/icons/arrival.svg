<svg width="23" height="20" viewBox="0 0 23 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M2.64073 7.60321C2.93741 7.73318 6.2387 9.87778 8.10128 10.3733C13.7041 11.8639 19.1336 13.2836 21.0194 13.2042C23.5229 13.0977 20.8609 10.2852 15.9558 9.21023C15.5608 9.12378 15.1526 9.03417 14.7349 8.94269L11.1796 5.60396L9.79647 5.30059L10.4708 8.00818C8.79362 7.64071 7.17849 7.28689 5.8628 6.99878L3.66508 2.84673L2.28186 2.54358L2.77724 6.32799C1.75287 6.18173 1.48732 7.09924 2.64073 7.60321Z" fill="#BDC0C8"/>
<rect x="1.50684" y="18.8387" width="12.2999" height="1.03704" fill="#BDC0C8"/>
</svg>
