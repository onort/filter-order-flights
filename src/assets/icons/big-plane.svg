<svg width="323" height="200" viewBox="0 0 323 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path opacity="0.05" fill-rule="evenodd" clip-rule="evenodd" d="M266.011 -27.5861C266.014 -27.5753 266.018 -27.5592 266.011 -27.5225C268.423 -19.5772 257.364 -3.24504 257.364 -3.24504L195.485 49.3344L211.797 204.81L194.972 217.13L155.402 108.3L141.11 94.9062L68.1102 156.737L64.7276 219.74L61.3487 223.41L35.6467 171.073L29.6805 171.31L28.2388 171.356L28.5482 169.819L29.7522 163.998L-17.7187 130.142L-13.5509 127.393L49.1637 134.286L122.017 72.2743L111.117 55.9924L10.1217 -0.697358L25 -15.3066L175.788 25.9988L237.696 -26.5366C237.696 -26.5366 263.151 -38.2615 266.011 -27.5861Z" fill="white"/>
</svg>
