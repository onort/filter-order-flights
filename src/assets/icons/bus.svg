<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M30.08 7.68H29.44V4.48C29.44 3.424 28.6672 2.17344 27.7229 1.70144C27.7229 1.70144 24.32 0 16 0C7.68 0 4.27744 1.70144 4.27744 1.70144C3.3328 2.17344 2.56 3.424 2.56 4.48V7.68H1.92C0.864 7.68 0 8.544 0 9.6V11.52C0 12.576 0.864 13.44 1.92 13.44H2.56V26.56C2.56 27.3917 3.09664 28.1018 3.84 28.367V29.44C3.84 30.848 4.992 32 6.4 32C7.808 32 8.96 30.848 8.96 29.44V28.48H23.04V29.44C23.04 30.848 24.192 32 25.6 32C27.008 32 28.16 30.848 28.16 29.44V28.367C28.9034 28.1018 29.44 27.3914 29.44 26.56V13.44H30.08C31.136 13.44 32 12.576 32 11.52V9.6C32 8.544 31.136 7.68 30.08 7.68ZM9.6 3.2C9.6 2.848 9.888 2.56 10.24 2.56H21.76C22.112 2.56 22.4 2.848 22.4 3.2V4.48C22.4 4.832 22.112 5.12 21.76 5.12H10.24C9.888 5.12 9.6 4.832 9.6 4.48V3.2ZM6.72 25.92C5.83616 25.92 5.12 25.2032 5.12 24.32C5.12 23.4368 5.83616 22.72 6.72 22.72C7.60352 22.72 8.32 23.4368 8.32 24.32C8.32 25.2032 7.60352 25.92 6.72 25.92ZM25.28 25.92C24.3962 25.92 23.68 25.2032 23.68 24.32C23.68 23.4368 24.3962 22.72 25.28 22.72C26.1638 22.72 26.88 23.4368 26.88 24.32C26.88 25.2032 26.1638 25.92 25.28 25.92ZM26.88 18.56C26.88 18.912 26.592 19.2 26.24 19.2H5.76C5.408 19.2 5.12 18.912 5.12 18.56V8.32C5.12 7.968 5.408 7.68 5.76 7.68H26.24C26.592 7.68 26.88 7.968 26.88 8.32V18.56Z" fill="#FF5A5F"/>
</svg>
