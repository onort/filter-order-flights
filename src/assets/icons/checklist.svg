<svg width="40" height="39" viewBox="0 0 40 39" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M9.24861 10.4413C9.05947 10.4413 8.86984 10.3701 8.72565 10.2277L6.72071 8.25386C6.43083 7.96867 6.43034 7.50555 6.71923 7.21987C7.00762 6.93371 7.47725 6.93322 7.76663 7.21841L9.24812 8.6775L12.4298 5.54482C12.7192 5.25964 13.1884 5.26012 13.4773 5.54629C13.7661 5.83196 13.7656 6.29509 13.4758 6.58027L9.77108 10.2282C9.62688 10.3701 9.43775 10.4413 9.24861 10.4413Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M9.24861 18.4889C9.05947 18.4889 8.86984 18.4178 8.72565 18.2754L6.72071 16.3015C6.43083 16.0163 6.43034 15.5532 6.71923 15.2675C7.00762 14.9814 7.47725 14.9809 7.76663 15.2661L9.24812 16.7252L12.4298 13.5925C12.7192 13.3068 13.1884 13.3078 13.4773 13.594C13.7661 13.8796 13.7656 14.3428 13.4758 14.6279L9.77108 18.2754C9.62688 18.4178 9.43775 18.4889 9.24861 18.4889Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M32.8665 9.33176H17.6236C17.2142 9.33176 16.8828 9.00465 16.8828 8.60051C16.8828 8.19638 17.2142 7.86926 17.6236 7.86926H32.866C33.2749 7.86926 33.6068 8.19638 33.6068 8.60051C33.6068 9.00465 33.2754 9.33176 32.8665 9.33176Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M32.8665 17.1994H17.6236C17.2142 17.1994 16.8828 16.8723 16.8828 16.4682C16.8828 16.0641 17.2142 15.7369 17.6236 15.7369H32.866C33.2749 15.7369 33.6068 16.0641 33.6068 16.4682C33.6068 16.8723 33.2754 17.1994 32.8665 17.1994Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M9.24861 33.4225C9.05947 33.4225 8.86984 33.3513 8.72565 33.2089L6.72071 31.2351C6.43083 30.9499 6.43034 30.4867 6.71923 30.2006C7.00762 29.9139 7.47725 29.9134 7.76663 30.1991L9.24812 31.6582L12.4298 28.525C12.7192 28.2399 13.1884 28.2403 13.4773 28.5265C13.7661 28.8127 13.7656 29.2753 13.4758 29.561L9.77108 33.2085C9.62688 33.3513 9.43775 33.4225 9.24861 33.4225Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M32.8665 32.1335H17.6236C17.2142 32.1335 16.8828 31.8059 16.8828 31.4023C16.8828 30.9986 17.2142 30.671 17.6236 30.671H32.866C33.2749 30.671 33.6068 30.9986 33.6068 31.4023C33.6068 31.8059 33.2754 32.1335 32.8665 32.1335Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M32.8665 25.0672H17.6236C17.2142 25.0672 16.8828 24.7396 16.8828 24.336C16.8828 23.9323 17.2142 23.6047 17.6236 23.6047H32.866C33.2749 23.6047 33.6068 23.9323 33.6068 24.336C33.6068 24.7396 33.2754 25.0672 32.8665 25.0672Z" fill="#FF5A5F"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M39.1766 38.6294H1.15187C0.742491 38.6294 0.411133 38.3018 0.411133 37.8982V0.848193C0.411133 0.444056 0.742491 0.116943 1.15187 0.116943H39.1766C39.5855 0.116943 39.9173 0.444056 39.9173 0.848193V37.8982C39.9173 38.3018 39.5855 38.6294 39.1766 38.6294ZM1.89261 37.1669H38.4358V1.57944H1.89261V37.1669Z" fill="#FF5A5F"/>
</svg>
