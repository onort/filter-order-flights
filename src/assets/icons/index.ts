import Arrival from "./arrival.svg"
import ArrivalDate from "./arrival-date.svg"
import ArrowRight from "./arrow-right.svg"
import Bus from "./bus.svg"
import CheckList from "./checklist.svg"
import Contact from "./contact.svg"
import Departure from "./departure.svg"
import DepartureDate from "./deparutre-date.svg"
import Facebook from "./facebook-icon.svg"
import Forward from "./forward.svg"
import GooglePlus from "./google-plus.svg"
import Help from "./help.svg"
import Home from "./home.svg"
import Location from "./location.svg"
import Luggage from "./luggage.svg"
import Map from "./map.svg"
import Meal from "./meal.svg"
import Menu from "./menu.svg"
import People from "./people.svg"
import Person from "./person.svg"
import Plane from "./plane.svg"
import Sale from "./sale.svg"
import SearchLoction from "./search-location.svg"
import Tick from "./tick.svg"
import Ticket from "./ticket.svg"
import Twitter from "./twitter-icon.svg"

export {
  Arrival,
  ArrivalDate,
  ArrowRight,
  Bus,
  CheckList,
  Contact,
  Departure,
  DepartureDate,
  Facebook,
  Forward,
  GooglePlus,
  Help,
  Home,
  Location,
  Luggage,
  Map,
  Meal,
  Menu,
  People,
  Person,
  Plane,
  Sale,
  SearchLoction,
  Tick,
  Ticket,
  Twitter
}
