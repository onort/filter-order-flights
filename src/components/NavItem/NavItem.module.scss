@import "../../styles/main.scss";

.item {
  display: flex;
  align-items: center;
  padding: map-get($spacers, normal) 0 map-get($spacers, large);
  margin-left: map-get($spacers, x-large);
  color: $grey2;
  cursor: pointer;

  &.active {
    color: $primary;
  }
}

.icon {
  margin: 0 map-get($spacers, small);
}

.title {
  font-family: $font-family-heading;
  font-weight: 600;
}
