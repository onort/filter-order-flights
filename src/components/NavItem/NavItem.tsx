import React from "react"
import cx from "classnames"

import styles from "./NavItem.module.scss"

interface Props {
  className?: string
  active?: boolean
  icon?: string
  title: string
}

const NavItem: React.SFC<Props> = props => {
  const classNames = cx(
    styles.item,
    { [styles.active]: props.active },
    props.className
  )
  return (
    <a className={classNames}>
      <img className={styles.icon} src={props.icon} alt={props.title} />
      <span className={styles.title}>{props.title}</span>
    </a>
  )
}

export default NavItem
