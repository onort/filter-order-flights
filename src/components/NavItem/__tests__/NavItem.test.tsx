import React from "react"
import { shallow } from "enzyme"

import NavItem from "../"
import { Home } from "../../../assets/icons"

describe("<NavItem />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<NavItem title="Home" icon={Home} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have a active classname when active prop has been passed", () => {
    const wrapper = shallow(<NavItem title="Home" icon={Home} active={true} />)
    expect(wrapper.find("a").hasClass("active")).toBe(true)
  })

  it("should have a custom classname", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <NavItem title="Home" icon={Home} className={testClassName} />
    )
    expect(wrapper.find("a").hasClass(testClassName)).toBe(true)
  })
})
