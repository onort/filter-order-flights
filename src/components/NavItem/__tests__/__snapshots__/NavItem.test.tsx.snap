// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavItem
    icon="home.svg"
    title="Home"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Home"
          className="icon"
          src="home.svg"
        />,
        <span
          className="title"
        >
          Home
        </span>,
      ],
      "className": "item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Home",
          "className": "icon",
          "src": "home.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Home",
          "className": "title",
        },
        "ref": null,
        "rendered": "Home",
        "type": "span",
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Home"
            className="icon"
            src="home.svg"
          />,
          <span
            className="title"
          >
            Home
          </span>,
        ],
        "className": "item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Home",
            "className": "icon",
            "src": "home.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "title",
          },
          "ref": null,
          "rendered": "Home",
          "type": "span",
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<NavItem /> should match snapshot 2`] = `"<a class=\\"item\\"><img class=\\"icon\\" src=\\"home.svg\\" alt=\\"Home\\"/><span class=\\"title\\">Home</span></a>"`;
